resources/tr.scala.template

```diff
@@ -6,15 +6,19 @@ import android.content.Context
 import android.content.res.{TypedArray,XmlResourceParser}
 import android.graphics.drawable.Drawable
 import android.os.Build
-import android.view.{View,ViewGroup,LayoutInflater}
+import android.view.{View,ViewGroup,LayoutInflater, MenuInflater, Menu, MenuItem}
 import android.view.animation.{Animation, AnimationUtils, Interpolator}
 import android.animation.{Animator, AnimatorInflater}
 
 import scala.annotation.implicitNotFound
 
+trait MenuItemExtractor { self: TypedRes[_] =>
+  def unapply(mi: MenuItem): Boolean = if (mi.getItemId == resid) true else false
+}
+
 case class TypedResource[A](id: Int) extends AnyVal
 case class TypedLayout[A](id: Int)%s
-case class TypedRes[A](resid: Int) extends AnyVal {
+case class TypedRes[A](resid: Int) extends MenuItemExtractor {
```

Comment:
Yes, a type-class approach is much cleaner. I didn't realize that scala's extractor mechanism works with type classes, but it seems to do. I updated the PR.